### DISCUSSION TITLE
Keystroke Message Flags

### DISCUSSION THREAD

**Discussion by HavenDV** [microsoft/CsWin32#519] [Q&A] [answered]
Is it possible to add Keystroke Message Flags using NativeMethods.txt?

https://docs.microsoft.com/en-us/windows/win32/inputdev/about-keyboard-input#keystroke-message-flags

**Comment by AArnott** [comment, marked as answer]
Sure. You should be able to ask for `KF_ALTDOWN` for example.

That produces this:

```cs
namespace Windows.Win32
{
	using global::System;
	using global::System.Diagnostics;
	using global::System.Runtime.CompilerServices;
	using global::System.Runtime.InteropServices;
	using global::System.Runtime.Versioning;
	using winmdroot = global::Windows.Win32;

	internal static partial class PInvoke
	{
		internal const uint KF_ALTDOWN = 8192U;
	}
}
```

**Reply by HavenDV** [reply]
Yes, thanks, I changed some constants in the code.
But also the code I maintain contains the following constants:
```cs
public const int KF_TOGGLED = 0x0001;
public const int KF_BYTE_UP = 0x80;
```
I'm assuming these are just hardcoded values for what [GetKeyState returns](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeystate#return-value). Do you happen to know if there are constants in the generator for this?

**Reply by AArnott** [reply]
I don't see those two in the latest metadata version. If they are in the C++ headers, please file a bug in the win32metadata repo
